### ISSUE TITLE
SET SESSION AUTHORIZATION unparsable in Redshift dialect

### ISSUE THREAD

**Issue by ramsespostnl** [sqlfluff/sqlfluff#6987]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I try to parse a SQL file with the `SET SESSION AUTHORIZATION` command inside, it fails to parse when I'm using the Redshift dialect. This should not happen, since this is valid Redshift SQL: <https://docs.aws.amazon.com/redshift/latest/dg/r_SET_SESSION_AUTHORIZATION.html>

### Expected Behaviour

This should be parseable.

I think this rule has been implemented in postgres dialect, but not (yet) in Redshift dialect.

https://github.com/sqlfluff/sqlfluff/blob/522d488c9505debd2a3706c0294180f39d9124f8/src/sqlfluff/dialects/dialect_postgres.py#L4972

### Observed Behaviour

`sqlfluff parse my_file.sql`

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Delimited: [<Ref: 'StatementSegment'>]>"
[L:  1, P:  1]      |        word:                                                 'set'
[L:  1, P:  4]      |        whitespace:                                           ' '
[L:  1, P:  5]      |        word:                                                 'session'
[L:  1, P: 12]      |        whitespace:                                           ' '
[L:  1, P: 13]      |        word:                                                 'authorization'
[L:  1, P: 26]      |        whitespace:                                           ' '
[L:  1, P: 27]      |        word:                                                 'my_user'
[L:  1, P: 34]      |        semicolon:                                            ';'
[L:  1, P: 35]      |        newline:                                              '\n'
[L:  2, P:  1]      |        newline:                                              '\n'
[L:  3, P:  1]      |        word:                                                 'reset'
[L:  3, P:  6]      |        whitespace:                                           ' '
[L:  3, P:  7]      |        word:                                                 'session'
[L:  3, P: 14]      |        whitespace:                                           ' '
[L:  3, P: 15]      |        word:                                                 'authorization'
[L:  3, P: 28]      |        semicolon:                                            ';'
[L:  3, P: 29]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'set session
                       | authorization my_user;\n\nrese...'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
```

### How to reproduce

Example, command run on a file like this:
`sqlfluff parse the_file_below.sql`

```SQL
set session authorization my_user;

reset session authorization;
```

### Dialect

Redshift.

### Version

sqlfluff, version 3.3.0

### Configuration

```
[sqlfluff]
dialect = redshift
sql_file_exts = .sql

```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ramsespostnl** [comment]
I've upgraded to `3.4.1` and then it did succesfully parsed. Closing the issue.
